opensearch-operator/pkg/builders/dashboards.go

```diff
@@ -182,6 +182,8 @@ func NewDashboardsSvcForCr(cr *opsterv1.OpenSearchCluster) *corev1.Service {
 			Labels:    labels,
 		},
 		Spec: corev1.ServiceSpec{
+			Type:                     cr.Spec.Dashboards.Service.Type,
```

Comment:
Again, in a case that a client will not insert `Spec.Dashboards.Service.Type` in his CRD, it will fail cause the service will not have a default type, we need to add at the begging of the dashboard creation something like that:
if Spec.Dashboards.Service.Type == nil {
           service_type := ClusterIp
else
           service_type := Spec.Dashboards.Service.Type

Spec: corev1.ServiceSpec{
	Type:      service_type

Reply:
@idanl21 no it will default to ClusterIP.  Check the CRD

Reply:
Hi @idanl21 @swoehrl-mw 

Do you still want the if statements in the code?  ClusterIP is the K8 default so it assumes this when nothing is set.  Plus I have set the default in the CRD so we don't need to pollute the code with if statements.  But happy to add them if you want me too.

Reply:
IMO the if is not needed in this case. I agree with @simonmacklin's argument. Although it might be sensible to add a comment to the code explaining this so anybody else understands it without needing context.